### PR TITLE
Replacing require.nil in model package

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Replacing require.nil in model package

#### Ticket Link
Not yet

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: model/access_test.go [hunk 1]
```diff
@@ -29,41 +29,41 @@ func TestAccessIsValid(t *testing.T) {
...
-	require.Error(t, ad.IsValid())
+	require.NotNil(t, ad.IsValid())
 
 	ad.RedirectUri = NewRandomString(28)
-	require.Error(t, ad.IsValid())
+	require.NotNil(t, ad.IsValid())
 
 	ad.RedirectUri = "http://example.com"
-	require.Error(t, ad.IsValid(), ad.IsValid())
+	require.Nil(t, ad.IsValid())
```
Comment (jespino) [inline model/access_test.go:68]: This was false-passing before, as `Error`, but now that is checking for `Nil` is having the expected behavior.

File: model/channel_member_test.go [hunk 2]
```diff
@@ -21,31 +21,31 @@ func TestChannelMemberJson(t *testing.T) {
...
-	require.Error(t, o.IsValid(), "should be invalid")
+	require.NotNil(t, o.IsValid(), "should be invalid")
 
 	o.NotifyProps["desktop"] = "123456789012345678901"
-	require.Error(t, o.IsValid(), "should be invalid")
+	require.NotNil(t, o.IsValid(), "should be invalid")
 
 	o.NotifyProps["desktop"] = CHANNEL_NOTIFY_ALL
-	require.Error(t, o.IsValid(), "should be invalid")
+	require.Nil(t, o.IsValid(), "should be valid")
```
Comment (jespino) [inline model/channel_member_test.go:39]: This was false-passing before, as `Error`, but now that is checking for `Nil` is having the expected behavior.

File: model/channel_member_test.go [hunk 3]
```diff
@@ -21,31 +21,31 @@ func TestChannelMemberJson(t *testing.T) {
...
 
 	o.NotifyProps["mark_unread"] = "123456789012345678901"
-	require.Error(t, o.IsValid(), "should be invalid")
+	require.NotNil(t, o.IsValid(), "should be invalid")
 
 	o.NotifyProps["mark_unread"] = CHANNEL_MARK_UNREAD_ALL
-	require.Error(t, o.IsValid(), "should be invalid")
+	require.Nil(t, o.IsValid(), "should be valid")
```
Comment (jespino) [inline model/channel_member_test.go:45]: This was false-passing before, as `Error`, but now that is checking for `Nil` is having the expected behavior.

File: model/channel_member_test.go [hunk 4]
```diff
@@ -21,31 +21,31 @@ func TestChannelMemberJson(t *testing.T) {
...
 
 	o.Roles = ""
-	require.Error(t, o.IsValid(), "should be invalid")
+	require.Nil(t, o.IsValid(), "should be invalid")
```
Comment (jespino) [inline model/channel_member_test.go:48]: This was false-passing before, as `Error`, but now that is checking for `Nil` is having the expected behavior.

File: model/channel_test.go [hunk 5]
```diff
@@ -53,41 +53,41 @@ func TestChannelPatch(t *testing.T) {
...
-	require.Error(t, o.IsValid())
+	require.NotNil(t, o.IsValid())
 
 	o.Type = "U"
-	require.Error(t, o.IsValid())
+	require.NotNil(t, o.IsValid())
 
 	o.Type = "P"
-	require.Error(t, o.IsValid())
+	require.Nil(t, o.IsValid())
```
Comment (jespino) [inline model/channel_test.go:81]: This was false-passing before, as `Error`, but now that is checking for `Nil` is having the expected behavior.

File: model/command_webhook_test.go [hunk 6]
```diff
@@ -46,7 +51,7 @@ func TestCommandWebhookIsValid(t *testing.T) {
 		err := h.IsValid()
 
 		if test.ExpectedError == "" {
-			assert.Error(t, err, "hook should be valid")
+			assert.Nil(t, err, "hook should be valid")
```
Comment (jespino) [inline model/command_webhook_test.go:54]: This was false-passing before, as `Error`, but now that is checking for `Nil` is having the expected behavior.

File: model/terms_of_service_test.go [hunk 7]
```diff
@@ -13,19 +13,19 @@ import (
...
-	assert.Error(t, s.IsValid(), "should be invalid")
+	assert.NotNil(t, s.IsValid(), "should be invalid")
 
 	s.CreateAt = GetMillis()
-	assert.Error(t, s.IsValid(), "should be invalid")
+	assert.NotNil(t, s.IsValid(), "should be invalid")
 
 	s.UserId = NewId()
-	assert.Error(t, s.IsValid(), "should be invalid")
+	assert.Nil(t, s.IsValid(), "should be valid")
```
Comment (jespino) [inline model/terms_of_service_test.go:25]: This was false-passing before, as `Error`, but now that is checking for `Nil` is having the expected behavior.

File: model/user_test.go [hunk 8]
```diff
@@ -126,11 +126,11 @@ func TestUserIsValid(t *testing.T) {
 	require.True(t, HasExpectedUserIsValidError(err, "nickname", user.Id), "expected user is valid error: %s", err.Error())
 
 	user.Nickname = strings.Repeat("a", 64)
-	require.Error(t, user.IsValid())
+	require.Nil(t, user.IsValid())
```
Comment (jespino) [inline model/user_test.go:129]: This was false-passing before, as `Error`, but now that is checking for `Nil` is having the expected behavior.

File: model/user_test.go [hunk 9]
```diff
@@ -126,11 +126,11 @@ func TestUserIsValid(t *testing.T) {
...
 
 	user.FirstName = ""
 	user.LastName = ""
-	require.Error(t, user.IsValid())
+	require.Nil(t, user.IsValid())
```
Comment (jespino) [inline model/user_test.go:133]: This was false-passing before, as `Error`, but now that is checking for `Nil` is having the expected behavior.

File: model/user_test.go [hunk 10]
```diff
@@ -143,7 +143,7 @@ func TestUserIsValid(t *testing.T) {
 
 	user.LastName = strings.Repeat("a", 64)
 	user.Position = strings.Repeat("a", 128)
-	require.Error(t, user.IsValid())
+	require.Nil(t, user.IsValid())
```
Comment (jespino) [inline model/user_test.go:146]: This was false-passing before, as `Error`, but now that is checking for `Nil` is having the expected behavior.

**Comment by jespino** [comment]
See the comments, some interesting cases where we were asserting super weird things about the `IsValid` method behavior.

**Review by agnivade — APPROVED** [review]
Nice, lots of mistakes fixed.

File: model/file_info_test.go [hunk 11]
```diff
@@ -77,13 +77,13 @@ func TestGetInfoForFile(t *testing.T) {
 	fakeFile := make([]byte, 1000)
 
 	pngFile, err := ioutil.ReadFile("../tests/test.png")
-	require.Nilf(t, err, "Failed to load test.png")
+	require.NoErrorf(t, err, "Failed to load test.png")
```
Comment (wiggin77) [inline model/file_info_test.go:80]:
```suggestion
	require.NoErrorf(t, err, "Failed to load test.png")
```
```suggestion
	require.NoError(t, err, "Failed to load test.png")
```

Comment (wiggin77) [inline model/file_info_test.go:80]:
```suggestion
	require.NoErrorf(t, err, "Failed to load test.png")
```
```suggestion
	require.NoError(t, err, "Failed to load test.png")
```

File: model/file_info_test.go [hunk 12]
```diff
@@ -77,13 +77,13 @@ func TestGetInfoForFile(t *testing.T) {
...
 
 	// base 64 encoded version of handtinywhite.gif from http://probablyprogramming.com/2009/03/15/the-tiniest-gif-ever
 	gifFile, _ := base64.StdEncoding.DecodeString("R0lGODlhAQABAIABAP///wAAACwAAAAAAQABAAACAkQBADs=")
 
 	animatedGifFile, err := ioutil.ReadFile("../tests/testgif.gif")
-	require.Nilf(t, err, "Failed to load testgif.gif")
+	require.NoErrorf(t, err, "Failed to load testgif.gif")
```
Comment (wiggin77) [inline model/file_info_test.go:86]:
```suggestion
	require.NoErrorf(t, err, "Failed to load testgif.gif")
```
```suggestion
	require.NoError(t, err, "Failed to load testgif.gif")
```

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:    2 minor suggestions.

**Pull request merged**